tools/elf_cls/src/main.rs

```diff
@@ -0,0 +1,282 @@
+//! Tool to prepare an ELF file with CLS sections.
+//!
+//! This involves replacing the `.cls` section's TLS flag with the CLS flag,
+//! updating all CLS symbols to have the CLS type, and updating various symbol
+//! values.
+//!
+//! Can be invoked on a single file
+//! ```
+//! elf_cls <arch> --file <path>
+//! ```
+//! or a directory of object files
+//! ```
+//! elf_cls <arch> --dir <path>
+//! ```
+#![feature(int_roundings)]
+
+use std::{
+    env,
+    ffi::OsStr,
+    fs::{self, File},
+    io::{Read, Seek, SeekFrom, Write},
+    path::{Path, PathBuf},
+};
+
+use goblin::{
+    container::{Container, Ctx, Endian},
+    elf::{
+        header::header64::Header,
+        section_header::SHT_SYMTAB,
+        sym::{Symtab, STT_HIOS, STT_LOOS},
+        SectionHeader,
+    },
+    elf64::{
+        section_header::{SHF_MASKOS, SHF_TLS},
+        sym::STT_TLS,
+    },
+    strtab::Strtab,
+};
+
+/// The flag identifying CLS sections.
+///
+/// This must be kept in sync with `mod_mgmt`.
+const CLS_SECTION_FLAG: u64 = 0x100000;
+
+const _: () = assert!(CLS_SECTION_FLAG & SHF_MASKOS as u64 == CLS_SECTION_FLAG);
+
+/// The flag identifying CLS symbols.
+///
+/// This must be kept in sync with `mod_mgmt`.
+const CLS_SYMBOL_TYPE: u8 = 0xa;
```

Comment:
can you define these two consts in `crate_metadata_serde` (which is intended for this purpose) such that this `elf_cls` tool can directly depend on that?

Then the `crate_metadata` (and thus also `mod_mgmt`) can re-export those symbols from `crate_metadata_serde` to have a single unified system-wide definition. This is similar to what we did with `SectionType` when you implemented nano_core serialization.